### PR TITLE
dev-java/gwt: bump to 2.9.0

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
Resolves bug #762619 and bug #716220

I opted to **not** close these bugs because I'm not sure if that's appropriate,
as the bugs still exist in the previous version of this ebuild.

Signed-off-by: Wolfgang E. Sanyer <ezzieyguywuf@gmail.com>

### CODE REVIEW

**Comment by ezzieyguywuf** [comment]
@juippis ping. As requested on #19865, this PR should address the issues you're having with gwt.

I'm curious to hear your take on the changes I made to the original ebuild - some of them are stylistic, but also e.g. I restrict the tests since they technically fail.

**Comment by ezzieyguywuf** [comment]
Please don't merge this yet, I'm trying to convince upstream to push a new version that will remove the need for a patch.

See here: https://gwt-review.googlesource.com/c/gwt/+/23240

File: dev-java/gwt/gwt-2.9.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		validation-api-1.0
+"
+
+PATCHES=( "${FILESDIR}/${PN}-2.9.0-jsinterop-classpath.patch" )
+
+src_prepare() {
+	default
+
+	java-pkg-2_src_prepare
+}
```
Comment (juippis) [inline dev-java/gwt/gwt-2.9.0.ebuild:60]: It's really weird the java eclass(es) don't handle applying patches, but reading them this does seem correct.

**Review by juippis — APPROVED** [review]
I'd be fine merging this, and waiting for a proper upstream release. @fordfrog let us know what to do, this blocks testing of a linked PR (check the opening comment).

Thanks, this works for me!

**Comment by ezzieyguywuf** [comment]
> I'd be fine merging this, and waiting for a proper upstream release. @fordfrog let us know what to do, this blocks testing of a linked PR (check the opening comment).
> 
> Thanks, this works for me!
> 

Sweet thanks for the review!

I'll switch off draft mode when I'm back at my comp, i can't figure out how to change it from the github android app, lol

File: dev-java/gwt/gwt-2.9.0.ebuild (hunk 1 above)
Comment (fordfrog) [reply dev-java/gwt/gwt-2.9.0.ebuild:60]: yeah, this caught me too :-)

**Comment by fordfrog** [comment]
i just tried to emerge it with `openjdk-11` and it fails because of missing `tools.jar`:
```
BUILD FAILED
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:92: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:16: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:42: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:16: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:88: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build_tools/build.xml:17: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build_tools/build.xml:11: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build_tools/doctool/build.xml:7: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/common.ant.xml:366: Cannot find /usr/lib64/openjdk-11/../lib/tools.jar; please use a JDK when building doc rather than a JRE.
```
though java 11 is still masked, this should be resolved so that the package does not fail when java 11 is unmasked.
there are generally two approaches.
1. find out what `tools.jar` is used for and whether it can be avoided
2. if 1. does not work, restrict `DEPEND` to `virtual/jdk:1.8`. this solution though might cause some issues in the future so 1. is preferred if possible.

File: dev-java/gwt/gwt-2.9.0.ebuild (hunk 1 above)
Comment (mm1ke) [reply dev-java/gwt/gwt-2.9.0.ebuild:60]: FYI, I've opened a bug about this behavior some time ago (https://bugs.gentoo.org/728814).
Some ebuilds had `PATCHES` set but didn't do any patching.

**Comment by fordfrog** [comment]
the compilation failed for me:
```
-filter.props:
     [echo] Branding as GWT version 2.9.0, GIT commit fatal: not a git repository (or any parent up to mount point /var/tmp)
     [echo] Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
     [copy] Copying 1 file to /var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build/out/dev/bin
    [touch] Creating /var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build/out/dev/sentinels/gwt-2.9.0-git-fatal: not a git repository (or any parent up to mount point /var/tmp)
    [touch] Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

BUILD FAILED
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:92: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:16: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build.xml:43: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/dev/build.xml:255: The following error occurred while executing this line:
/var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/dev/build.xml:183: Could not create /var/tmp/portage/dev-java/gwt-2.9.0/work/gwt-2.9.0/build/out/dev/sentinels/gwt-2.9.0-git-fatal: not a git repository (or any parent up to mount point /var/tmp)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

**Comment by ezzieyguywuf** [comment]
> the compilation failed for me:

I fixed this with a new patch

> also, though unrelated, it would be good to try whether it compiles against the latest slot of `asm` and switch to that if it is fine.

I checked - it works with the latest `asm`. I had to modify `EANT_GENTOO_CLASSPATH` in order to build with the latest slot - should I leave this value as-is, or do you want me to set it back to `asm-4` so that it works with the latest stable `asm`?

File: dev-java/gwt/gwt-2.9.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+RESTRICT="mirror test"
+SLOT="2.8"
+LICENSE="Apache-2.0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+CDEPEND="
+	dev-java/json:0
+	dev-java/guava:20
+	dev-java/rhino:1.6
+	>=dev-java/asm-5.0.3:4
```
Comment (fordfrog) [inline dev-java/gwt/gwt-2.9.0.ebuild:30]: this also needs to be updated to `dev-java/asm:9` so that it pulls in the correct dependency.

Comment (ezzieyguywuf) [reply dev-java/gwt/gwt-2.9.0.ebuild]: Fixed.

**Comment by fordfrog** [comment]
> > the compilation failed for me:
> 
> I fixed this with a new patch

it seems you forgot to add the patch:
```
 * Applying gwt-2.9.0-remove-git-usage.patch ...
/var/tmp/portage/dev-java/gwt-2.9.0/temp/environment: line 601: /var/tmp/portage/dev-java/gwt-2.9.0/files/gwt-2.9.0-remove-git-usage.patch: No such file or directory
/var/tmp/portage/dev-java/gwt-2.9.0/temp/environment: line 604: /var/tmp/portage/dev-java/gwt-2.9.0/files/gwt-2.9.0-remove-git-usage.patch: No such file or directory
```

**Comment by fordfrog** [comment]
@ezzieyguywuf merged, thanks! if you encounter any issues with gwt when testing rstudio, let me know, i saw some suspicious things during the build but it might work as it is.

**Pull request closed without merging**